ts/types.ts

```diff
@@ -30,7 +30,16 @@ export type TestResult = {
   error?: string
 };
 
-export type CompilerOpts = {
+type CommonOpts = {
```

Comment:
I believe we can switch back to just using one single type, `CompilerOpts`. Something like:
```typescript
type CompilerOpts = {
  consoleLog: (message: string) => void,
  version: () => void,
  whitelistCode?: { [key: string]: string }
}
```
`jsonPath` is not necessary since we're gonna do all file fetching/loading separate from EJS.

Reply:
Pushed changes; take another look.